### ISSUE TITLE
Shuffles in choice text are re-shuffled upon making a choice

### ISSUE THREAD

**Issue by hoverbird** [inkle/ink#240]
I'm writing a story with a lot of branching, semi-random content— including the text players select when making a choice. There's this behavior (at least in Inky) where a shuffle gets evaluated twice— once on displaying the choice, and again when inserting the choice text into the transcript. 

This is super weird to the player, as they are instructing their character to say one thing, only to have them say a variant. This feels like a bug to me... although I could imagine use cases for some things (like inline functions) where it might make sense to evaluate them on display and on player action, shuffles aren't one of those things.

**Comment by joningold** [comment]
This behaviour is super unhelpful! I tend to work around it using temps -
assign the text to a temp then print it - but this is kinda nasty.

Would be nice to fix!

Jon

**Comment by joethephish** [comment]
Yeah I agree... unfortunately it's quite hard to fix because of the way the choice content both has to be capable of being dynamic and repeat for the choice text generation as well as for the actual content generation.

It would definitely be good to fix in future but I'm sorry to say it's unlikely to be any time soon, sorry :-(

**Comment by hoverbird** [comment]
I can see how, each time a choice is rendered, you'd want to reevaluate it— for instance, when a player passes through the same node twice, for example. But in between rendering the choice and then displaying its content, why does that need to be evaluated twice?... is it because the part after the `[]` could contain code that you don't want to execute until it's selected?

**Comment by joethephish** [comment]
So, in the structure of the choice:

`* A [B] C`

C is correctly rendered only once, same for B. And yeah, we ought to be able to cache the raw string for A, but right now that's simply not how it's set up, sorry! It would definitely be good to refactor and rebuild choices to be a bit simpler internally and do this kind of caching.

Choices were designed from the ground up to just be simple divert targets. So when diverting somewhere, knowing that it's as a result of a choice, and getting hold of that cached piece of text isn't as easy as simply re-calling an internal piece of content (A).

**Comment by hoverbird** [comment]
Ahh, thanks for the explanation! I can understand why special-casing Choices to be evaluated differently from other divert targets would be a pretty major refactor.
